imports/api/users.js

```diff
@@ -263,7 +263,7 @@ Meteor.methods({
     if (!Meteor.user().hasGreaterRole('professor')) throw new Meteor.Error('invalid-permissions', 'Invalid permissions')
```

Comment:
If this gets called when sharing a question, I assume that a student could not share a question. We also don't want students to be able to query the user db.
To share a question, make a method that shares by email address. In the method, check that there exist a user with that email address, otherwise return an error "no user with that email address". 
When a user shares a question, what should happen to courseID? E.g. can you share a question with a student not in your course? Perhaps courseId should remain if the receiving student/prof is in the course, otherwise, in the list of questions shared with a user, the user has to select one of their courses when they move the question to their library?

Reply:
Right now, sharing a question does not change the question's courseId. However, in the questions.sharedWithUser library the subscription calls all questions that are shared with the user (Not just in the course the user is currently in). Then, when the user copies a question to their library the question's courseId is set to the course the user is currently in. This allows users to share questions with anybody via email. Perhaps this should be changed for students if we want them only to be able to share questions with only students in their course?